autoload/cmake4vim.vim

```diff
@@ -227,4 +227,39 @@ function! cmake4vim#RunTarget(bang, ...) abort
         call utils#common#Warning(v:errmsg)
     endif
 endfunction
+
+" Complete CCMake window modes
+function! cmake4vim#CompleteCCMakeModes(arg_lead, cmd_line, cursor_pos) abort
+    let l:modes = ['h', 'v', 't']
+    return join(l:modes, "\n")
+endfunction
+
+" Open CCMake window
+function! cmake4vim#CCMake(...) abort
+    " Supported modes:
+    " * empty, h - Open ccmake in horizontal split
+    " * v - Open ccmake in vertical split
+    " * t - Open ccmake in new tab
+    let l:mode = ''
+    if exists('a:1') && a:1 !=# ''
```

Comment:
You can check `if a:0` instead of `exists('a:1')`.

`a:0` is the number of variable arguments and it's `0` if the function was called with no arguments.

So this could actually be like this:
```
let l:mode = a:0 ? a:1 : 'h'
if has('nvim')
...
```

Reply:
Good catch!